front/pages/no-workspace.tsx

```diff
@@ -44,8 +45,18 @@ async function fetchRevokedWorkspace(
 ): Promise<Workspace | null> {
   // TODO(@fontanierh): this doesn't look very solid as it will start to behave
   // weirdly if a user has multiple revoked memberships.
+  const userRes = await UserResource.fetchByModelId(user.id);
+
+  if (!userRes) {
```

Comment:
This feels wrong. Do we want to keep user type as signature or do we want the auth to return a resource? cc @flvndvd

Reply:
For now, we should still return `UserType` from the Authenticator. Specifically here, we should never have Resources used in `pages` code. It seems way too easy to break something when trying to load server dependencies when running front-end code.

Reply:
Not too sure what you're saying there - I shouldn't pulling the user resource?
Pb being that the getLatestMemberships expects a userResource. Should I change it so that it accepts both UserType and UserResource?